### PR TITLE
i18n: Add custom translation file and doc

### PR DESCRIPTION
**Pull request by matheusd**
This allows loading custom translation files for the main UI (.json) and adds a doc explaining the translation process for communitity members.

A development build with the ability to import the custom translation files is available below. [Preview the doc](https://github.com/matheusd/decrediton/blob/community-i18n/app/i18n/community_translators.md) for community translators.

**Warning**: Ideally only use those binaries on a VM to ensure it doesn't mess around on your existing install.

https://github.com/matheusd/decred-weekly-builds/releases/tag/v20211013150209

### CODE REVIEW

File: app/i18n/community_translators.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+The next sections detail both translation flows.
+
+## Basic UI Translation
+
+Basic UI translation can be done once an RC (release candidate) version of Decrediton is released.
+
+Translators have to fulfill the following requirements:
+
+- Create a [Github](https://github.com) account
```
Comment (xaur) [inline app/i18n/community_translators.md:24]: Small nit: Git**H**ub, across the entire doc.

File: app/i18n/community_translators.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+- Install [git software](https://guides.github.com/introduction/git-handbook/)
+- Download the latest released Decrediton RC version
+- Follow the [#decrediton](https://matrix.to/#/#decrediton:decred.org) matrix chat room for announcements
```
Comment (xaur) [inline app/i18n/community_translators.md:27]: Optionally, also mention `#translations` chat room.

File: app/i18n/community_translators.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+- `<lang>.json` (this file contains the latest translation for the given language).
+
+Use the translation UI to fill any missing strings. The UI should be pretty self-explanatory.
+
+### Test and Complete the Translation
+
+During the translation process, the progress can be checked by saving the translation file somewhere and then loading it into Decrediton,
+using the `View -> Load Custom Translation` menu item.
+
+After translating all strings, replace the existing `lang.json` file in the Decrediton repository when all strings have been properly translated.
```
Comment (xaur) [inline app/i18n/community_translators.md:79]: `<lang>.json`

File: app/i18n/community_translators.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+### Use the Translation UI
+
+Open the file `decrediton-install-dir/app/i18n/translator.html` in a browser. This can be done by opening a modern browser, typing `file:///` in the address bar, then navigating to the dir where Decrediton source code as checked out.
+
+That file contains a very simple UI for working with Decrediton translation files.
+
+Load the following files, all located in the checked out copy of the repository, in the `app/i18n/translations` subdir:
+
+- `previous_original.json` (this file contains the original strings of the _previous_ Decrediton version)
```
Comment (xaur) [inline app/i18n/community_translators.md:68]: It is not clear from this doc how this file is involved. But I haven't checked `translator.html`, perhaps it is explained there. If so, ignore this comment.

**Review by xaur — COMMENTED** [review]
Great to see this! Just a few nits for `community_translations.md` (not commenting on other files).

File: app/i18n/community_translators.md (hunk 2 above)
Comment (matheusd) [reply app/i18n/community_translators.md:27]: Changed to `#translations` exclusively. I believe that's the appropriate channel for translators to decred (`#decrediton` might be too noisy).

File: app/i18n/community_translators.md (hunk 4 above)
Comment (matheusd) [reply app/i18n/community_translators.md:68]: It's used to determine source strings that changed between releases (hypothetical example, `app.title` changed from `Decrediton` to `Here's Decrediton`). 

There aren't really "explanations" in the translator file (it's pretty rough for the moment) but it shouldn't really matter as long as the translator loads the file.

**Pull request merged**